src/main/java/com/redhat/devtools/lsp4ij/features/documentation/markdown/LSPLinkResolver.java

```diff
@@ -0,0 +1,114 @@
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.documentation.markdown;
+
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import com.redhat.devtools.lsp4ij.features.documentation.MarkdownConverter;
+import com.vladsch.flexmark.ast.Image;
+import com.vladsch.flexmark.ast.Link;
+import com.vladsch.flexmark.ast.Reference;
+import com.vladsch.flexmark.html.LinkResolver;
+import com.vladsch.flexmark.html.LinkResolverFactory;
+import com.vladsch.flexmark.html.renderer.LinkResolverBasicContext;
+import com.vladsch.flexmark.html.renderer.LinkStatus;
+import com.vladsch.flexmark.html.renderer.ResolvedLink;
+import com.vladsch.flexmark.util.ast.Node;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * Custom link resolver used to resolve relative path by using the {@link com.intellij.psi.PsiFile}
+ * path which triggers the MarkDown converter for hover and completion documentation.
+ */
+public class LSPLinkResolver implements LinkResolver {
+
+    private final Path baseDir;
+
+    private enum FileUrlKind {
+        RELATIVE,
+        ABSOLUTE,
+        NONE;
+    }
+
+    public LSPLinkResolver(LinkResolverBasicContext context) {
+        this.baseDir = MarkdownConverter.BASE_DIR_CONTEXT.get(context.getOptions());
+    }
+
+    @Override
+    public @NotNull ResolvedLink resolveLink(@NotNull Node node, @NotNull LinkResolverBasicContext context, @NotNull ResolvedLink link) {
+        if (node instanceof Image || node instanceof Link || node instanceof Reference) {
+            String url = link.getUrl();
+            FileUrlKind fileUrlKind = getFileUrlKind(url);
+            if (baseDir != null && fileUrlKind == FileUrlKind.RELATIVE) {
+                String position = "";
+                int hashIndex= url.indexOf("#");
+                if (hashIndex != -1) {
+                    position = url.substring(hashIndex, url.length());
+                    url = url.substring(0, hashIndex);
+                }
+                try {
+                    File resolvedFile = baseDir.resolve(url).toFile();
+                    String resolvedUri = LSPIJUtils.toUri(resolvedFile).toASCIIString() + position;
+                    return link.withStatus(LinkStatus.VALID)
+                            .withUrl(resolvedUri);
+                }
+                catch(Exception e) {
+
+                }
+            }
+        }
+        return link;
+    }
+
+
+    private static FileUrlKind getFileUrlKind(String url) {
+        int index = url.indexOf("://");
+        if (index == -1) {

```

Comment:
it seems /Users/fred/foo/bar/file.txt would be -wrongly- considered a relative path.

Reply:
Good catch! fixed with test.